### PR TITLE
golines 0.11.0 (new formula)

### PR DESCRIPTION
**Pull request by darkenmay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by dawidd6 — DISMISSED** [review]

File: Formula/golines.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Golines < Formula
+  desc "Golang formatter that fixes long lines"
+  homepage "https://github.com/segmentio/golines"
+  url "https://github.com/segmentio/golines/archive/v0.11.0.tar.gz"
+  sha256 "d7336fbddb045bd2448629c4b8ef5ab2dc6136e71a795b6fdd596066bc00adc0"
+  license "MIT"
+
+  depends_on "go"
```
Comment (ZhongRuoyu) [inline Formula/golines.rb:8]: It should be sufficient to make it a build dependency:

```suggestion
  depends_on "go" => :build
```

File: Formula/golines.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+class Golines < Formula
+  desc "Golang formatter that fixes long lines"
+  homepage "https://github.com/segmentio/golines"
+  url "https://github.com/segmentio/golines/archive/v0.11.0.tar.gz"
```
Comment (ZhongRuoyu) [inline Formula/golines.rb:4]:
```suggestion
  url "https://github.com/segmentio/golines/archive/refs/tags/v0.11.0.tar.gz"
```

File: Formula/golines.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  sha256 "d7336fbddb045bd2448629c4b8ef5ab2dc6136e71a795b6fdd596066bc00adc0"
+  license "MIT"
+
```
Comment (ZhongRuoyu) [inline Formula/golines.rb:7]: Let's support head builds, too.

```suggestion
  head "https://github.com/segmentio/golines.git", branch: "master"

```

File: Formula/golines.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      package main
+
+      var strings = []string{"foo", "bar", "baz"}
+    EOS
+    (testpath/"expected.go").write <<~EOS
+      package main
+
+      var strings = []string{\n\t"foo",\n\t"bar",\n\t"baz",\n}
+    EOS
+    assert_match shell_output("#{bin}/golines --max-len=30 given.go"), (testpath/"expected.go").read
```
Comment (ZhongRuoyu) [inline Formula/golines.rb:25]:
```suggestion
    assert_equal shell_output("#{bin}/golines --max-len=30 given.go"), (testpath/"expected.go").read
```

**Review by SMillerDev — APPROVED** [review]

**Review by ZhongRuoyu — APPROVED** [review]
Thanks for your first contribution to `homebrew/core`, @darkenmay!

**Pull request closed without merging**